### ISSUE TITLE
Low accuracy for Qwen-2.5-7b-instruct model

### ISSUE THREAD

**Issue by sravan500** [NVIDIA/NeMo#13969]
**Describe the bug**
I have tested finetuning with qwen2.5-7b-instruct model.
with the scripts:
python3 examples/llm/finetune/automodel.py

All with default values,
8 H100 GPUS
Batch size=1
The training completed in less than 4 hours.

But accuracy of the model seems to be very low.
Did anyone face this issue?

**Steps/Code to reproduce bug**

Please list *minimal* steps or code snippet for us to be able to reproduce the bug.

A  helpful guide on on how to craft a minimal bug report  http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports. 

**Expected behavior**

**Environment overview (please complete the following information)**

 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider - AWS, Azure, GCP, Collab)]
 - Method of NeMo install: [pip install or from source]. Please specify exact commands you used to install.
 - If method of install is [Docker], provide `docker pull` & `docker run` commands used

**Environment details**

If NVIDIA docker image is used you don't need to specify these.
Otherwise, please provide:
- OS version
- PyTorch version
- Python version

**Additional context**

Example: GPU model

**Comment by abhinavg4** [comment]
Hi can you please provide more details on your training and testing setup?

**Comment by sravan500** [comment]
@abhinavg4
Thank you for reaching out.I was waiting for the response.

I have been working on finetuning several models using peft.
I have executed the code from:

/examples/llm/finetune/automodel.py
as:
 python3 automodel.py

After fine-tuning, I got the HF checkpoints, which I merged with the base model.
The same SQuAD dataset is used with a learning rate of 1e-5 for the Llama-3.2-3b-instruct model for finetuning.

I am evaluating with custom eval code, which shows accuracy improvement. Baseline 50% to Finetuned 60%.

But we also have our custom SFT codebase for training using SeqToSeqTrainer, which gives higher accuracy than using AutoModel for fine-tuning.

I request you to kindly answer my below questions:

1. After fine-tuning with the SQuAD dataset, do you have any eval scripts to compare accuracy differences?
I use a validation set. Can we evaluate with any other dataset?

2. Training is faster. What are the optimizations used? Please provide.

3. Training seems to be faster, but training loss doesn't come down in a gradual manner.

Please find the screenshot:

![Image](https://github.com/user-attachments/assets/f366a9f2-553d-431b-a645-f25f9570eecf)

**Comment by cuichenx** [comment]
Hi @sravan500 

1. Squad changes the output distribution quite a bit, so you should use a squad validation/test set for evaluation, not any other dataset. Usually you would run generation and compare the exact match accuracy.
2. 3. @akoumpa could you help here?

**Comment by sravan500** [comment]
Hi @akoumpa,
I have  finetuned a model using automodel example scripts.
Which is the right metric to be used for evaluation.As evaluation is task specific.

Base model: Also contains the ground truth.
Predictions also conatin the ground truth.

Do we need to evaluate the "correctness" or the format of the output??

Kindly find an example:
ground_truth"American Football Conference"

Finetuned model- Llama response:
"American Football Conference. Okay, so I've got this question about the Super Bowl, specifically asking what the AFC stands for. I remember that the AFC is the American Football Conference, but I need to make sure I'm accurate here. First, I should recall what the AFC is. It's one of the two main conferences in the NFL. The NFL is divided into conferences to organize the teams and scheduling. The AFC includes teams based in the United States and parts of Canada, while the NFC includes teams from Canada and the United States on the opposite coast. The AFC teams are: - Denver Broncos - Kansas City Chiefs - Houston Texans - Cleveland Browns - Cincinnati Bengals - Pittsburgh Steelers - New England Patriots - Atlanta Falcons"

Finetuned model Qwen's response:
"National Football Conference"

Kindly tell me which metric to use for evaluation?
The above mean that Llama didnt learn to respons in certain way?
Also is conciseness is considered or correctness?

Will be waiting for your response.
